src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetFixedSupply.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import "../../contracts/permissions/PauserRegistry.sol";
+import "../../contracts/interfaces/IStrategyManager.sol";
+import "../../contracts/interfaces/IServiceManager.sol";
+import "../../contracts/interfaces/IVoteWeigher.sol";
+import "../../contracts/middleware/VoteWeigherBase.sol";
+
+import "../harnesses/VoteWeigherBaseHarness.sol";
+
+import "../mocks/StrategyManagerMock.sol";
+import "../mocks/OwnableMock.sol";
+import "../mocks/DelegationMock.sol";
+
+import "forge-std/Test.sol";
+
+contract VoteWeigherBaseUnitTests is Test {
+
+    Vm cheats = Vm(HEVM_ADDRESS);
+
+    ProxyAdmin public proxyAdmin;
+    PauserRegistry public pauserRegistry;
+    IStrategyManager public strategyManager;
+
+    address serviceManagerOwner;
+    IServiceManager public serviceManager;
+
+    DelegationMock delegationMock;
+
+    VoteWeigherBaseHarness public voteWeigher;
+
+    VoteWeigherBaseHarness public voteWeigherImplementation;
+
+    address public pauser = address(555);
+    address public unpauser = address(999);
+
+    uint256 initialSupply = 1e36;
+    address initialOwner = address(this);
+
+    /// @notice emitted when a new quorum is created
+    event QuorumCreated(uint8 indexed quorumNumber);
+    /// @notice emitted when `strategy` has been added to the array at `strategiesConsideredAndMultipliers[quorumNumber]` with the `multiplier`
+    event StrategyAddedToQuorum(uint8 indexed quorumNumber, IStrategy strategy, uint96 multiplier);
+    /// @notice emitted when `strategy` has removed from the array at `strategiesConsideredAndMultipliers[quorumNumber]`
+    event StrategyRemovedFromQuorum(uint8 indexed quorumNumber, IStrategy strategy);
+    /// @notice emitted when `strategy` has its `multiplier` updated in the array at `strategiesConsideredAndMultipliers[quorumNumber]`
+    event StrategyMultiplierUpdated(uint8 indexed quorumNumber, IStrategy strategy, uint256 multiplier);
+
+    function setUp() virtual public {
+        proxyAdmin = new ProxyAdmin();
+
+        pauserRegistry = new PauserRegistry(pauser, unpauser);
+
+        StrategyManagerMock strategyManagerMock = new StrategyManagerMock();
+        delegationMock = new DelegationMock();
+        strategyManagerMock.setAddresses(
+            delegationMock,
+            IEigenPodManager(address(uint160(uint256(keccak256(abi.encodePacked("eigenPodManager")))))),
+            ISlasher(address(uint160(uint256(keccak256(abi.encodePacked("slasher"))))))
+        );
+
+        strategyManager = IStrategyManager(address(strategyManagerMock));
+
+        // make the serviceManagerOwner the owner of the serviceManager contract
+        cheats.prank(serviceManagerOwner);
+        serviceManager = IServiceManager(address(new OwnableMock()));
```

Comment:
probably want to `cheats.stopPrank()` after this call?

Reply:
oops nevermind, I'm used to using `startPrank` and `stopPrank` instead of just `prank`.